packages/typespec-ts/test/modularIntegration/modelPropertyAdditional.spec.ts

```diff
@@ -0,0 +1,733 @@
+        prop: { state: "ok" }
+      }
+    });
+    assert.equal(result, undefined);
+  });
+  it("should get spread Different Model Array value", async () => {
+    const result = await client.spreadDifferentModelArray.get();
+    assert.deepEqual(result, {
+      knownProp: "abc",
+      additionalProperties: {
+        prop: [{ state: "ok" }, { state: "ok" }]
+      }
+    });
+  });
+  it("should put spread Different Model Array value", async () => {
+    const result = await client.spreadDifferentModelArray.put({
+      knownProp: "abc",
+      additionalProperties: {
+        prop: [{ state: "ok" }, { state: "ok" }]
+      }
+    });
+    assert.equal(result, undefined);
+  });
+  it("should get spread Different String value", async () => {
+    const result = await client.spreadDifferentString.get();
+    assert.deepEqual(result, {
+      id: 43.125,
+      additionalProperties: {
+        prop: "abc"
+      }
+    });
+  });
+  it("should put spread Different String value", async () => {
+    const result = await client.spreadDifferentString.put({
+      id: 43.125,
+      additionalProperties: {
+        prop: "abc"
+      }
+    });
+    assert.equal(result, undefined);
+  });
+  it("should get spread Float value", async () => {
+    const result = await client.spreadFloat.get();
+    assert.deepEqual(result, {
+      id: 43.125,
+      additionalProperties: {
+        prop: 43.125
+      }
+    });
+  });
+  it("should put spread Float value", async () => {
+    const result = await client.spreadFloat.put({
+      id: 43.125,
+      additionalProperties: {
+        prop: 43.125
+      }
+    });
+    assert.equal(result, undefined);
+  });
+  it("should get spread Model value", async () => {
+    const result = await client.spreadModel.get();
+    assert.deepEqual(result, {
+      knownProp: { state: "ok" },
+      additionalProperties: {
+        prop: { state: "ok" }
+      }
+    });
+  });
+  it("should put spread Model value", async () => {
+    const result = await client.spreadModel.put({
+      knownProp: { state: "ok" },
+      additionalProperties: {
+        prop: { state: "ok" }
+      }
+    });
+    assert.equal(result, undefined);
+  });
+  it("should get spread Model Array value", async () => {
+    const result = await client.spreadModelArray.get();
+    assert.deepEqual(result, {
+      knownProp: [{ state: "ok" }, { state: "ok" }],
+      additionalProperties: {
+        prop: [{ state: "ok" }, { state: "ok" }]
+      }
+    });
+  });
+  it("should put spread Model Array value", async () => {
+    const result = await client.spreadModelArray.put({
+      knownProp: [{ state: "ok" }, { state: "ok" }],
+      additionalProperties: {
+        prop: [{ state: "ok" }, { state: "ok" }]
+      }
+    });
+    assert.equal(result, undefined);
+  });
+  // TODO: https://github.com/Azure/autorest.typescript/issues/3122
+  it.skip("should get spread Record Discriminated Union value", async () => {
```

Comment:
any complex ser/deser logic should be covered by followup issue https://github.com/Azure/autorest.typescript/issues/3122.